apollo-portal/src/main/resources/static/scripts/controller/IndexController.js

```diff
@@ -114,11 +126,37 @@ function IndexController($scope, $window, $translate, toastr, AppUtil, AppServic
                             app.favoriteId = favorite.id;
                             $scope.favorites.push(app);
                         });
-
                     });
             })
     }
 
+    function getPublicNamespaces() {
+        var size = 10;
+        NamespaceService.find_public_namespaces()
+            .then(function (result) {
+                $scope.allPublicNamespaces = result;
+                morePublicNamespace();
+                var selectResult = [];
+                angular.forEach(result,function (app) {
+                    selectResult.push({
+                        id: app.appId,
+                        text: app.appId + ' / ' + app.name
+                    })
+                });
+                $('#public-name-spaces-search-list').select2({
+                    data: selectResult,
+                    matcher: matchStart,
+                    allowClear:true,
```

Comment:
I think we don't need to write a custom matcher implementation? Is it ok to use the built-in matcher?
Also I think `allowClear` is not necessary here? Since when user selects some option, we will jump the target page.
```suggestion
```